arbor/label_resolution.hpp

```diff
@@ -3,12 +3,19 @@
 #include <unordered_map>
 #include <vector>
 
-#include "util/partition.hpp"
-
+#include <arbor/arbexcept.hpp>
 #include <arbor/common_types.hpp>
+#include <arbor/util/expected.hpp>
+
+#include "util/partition.hpp"
 
 namespace arb {
 
+struct resolution_error: arb::arbor_exception {
+    resolution_error(const std::string& msg);
+};
+using lid_hopefully = arb::util::expected<cell_lid_type , resolution_error>;
```

Comment:
`resolution_error` needn't be derived from an exception class; it can just be a wrapper around a string (or other data). Or even just a bare string. As we don't throw them, inheriting from `arbor_exception` is misleading.